### ISSUE TITLE
Flavors: Add `mmark` flavor support

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#401]
## Problem

rumdl flavors let users lint Markdown variants without false positives. `mmark` (Mmark) is a Markdown superset used for IETF-style documents and adds syntax that `standard` does not recognize, which leads to noisy lint results.

## Request

Add a new flavor: `mmark`, based on `standard`, with flavor-specific handling for Mmark constructs.

Config should match existing flavors:

```toml
[global]
flavor = "mmark"

[per-file-flavor]
"**/*.mmark" = "mmark"
```

Auto-detect `**/*.mmark` as `mmark` when no flavor is configured.

## Syntax to support

At minimum, treat these as valid Mmark syntax so they do not trigger spurious rule violations:

### Title block (TOML fenced by `%`)

```text
%%%
title = "Example"
%%%
```

### Special section headings

```text
.# Abstract
```

### Include directives

```text
{{path/to/file}}[3,10]
<{{path/to/file.go}}[/START/,/END/]
```

### Document divisions

```text
{frontmatter}
{mainmatter}
{backmatter}
```

### Captions for blocks

```text
Table: Caption text.
Figure: Caption text. {#id}
Quote: Caption text.
```

### Figures and subfigures fence

```text
!---
![Alt](img.png)
!---
```

### Block attributes

```text
{#id .class key="value"}
```

### Citations

```text
[@RFC2535]
[@!RFC1034;@RFC1035]
```

### XML references (inline fragments)

```text
<reference anchor='pandoc' target='https://example.com/'>
```

## Suggested rule adjustments

Initial set of rules to adjust for this flavor (can be expanded based on real-world examples):

- `MD041`: ignore a leading title block and allow `.#[space]...` special section headers before the first normal heading.
- `MD033`: allow Mmark XML reference fragments (at least `<reference ...>` blocks).
- `MD042`, `MD051`, `MD052`: do not treat citations, include directives, captions, or figure fences as links or reference definitions.

## Acceptance criteria

- `mmark` is a valid `flavor` value and is documented in `docs/flavors.md`.
- Add `docs/flavors/mmark.md` describing supported constructs and rule impacts.
- Auto-detection includes `.mmark`.
- Add fixtures/tests that cover the constructs above and demonstrate reduced false positives vs `standard`.

## References

- https://github.com/rvben/rumdl/blob/main/docs/flavors.md
- https://mmark.miek.nl/post/syntax/
- https://github.com/mmarkdown/mmark/blob/master/Syntax.md
- https://github.com/mmarkdown/mmark/blob/master/README.md
- https://github.com/mmarkdown/mmark/blob/master/FAQ.md

**Comment by rvben** [comment]
Thanks for the thorough write-up, @pygarap. We're going to defer mmark support for now. The IETF document authoring community is small, and kramdown-rfc appears to be the more widely used tool in that space — rumdl already supports the kramdown flavor which covers some overlap.

For mmark files, you can use config overrides to suppress the false positives:
```toml
[per-file-ignores]
"**/*.mmark" = ["MD033", "MD041", "MD042", "MD051", "MD052"]
```

If there's growing demand we'll reconsider.
